src/sankey.js

```diff
@@ -0,0 +1,152 @@
+import React, {PropTypes, Component, cloneElement} from 'react';
+import shallowCompare from 'react-addons-shallow-compare';
+import linkKey from './utils/link-key';
+import getWidthSums from './utils/get-width-sums';
+import separateChildrenByType from './utils/separate-children-by-type';
+import findColumns from './utils/find-columns';
+import configureNodeCoordinates from './utils/configure-node-coordinates';
+import getAnchorFromRectangleNodes from './utils/get-anchor-from-rectangle-nodes';
+
+const {number} = PropTypes;
+export default class Sankey extends Component {
+
+  static propTypes = {
+    /** width of svg figure */
+    width: number.isRequired,
+    /** height of svg figure */
+    height: number.isRequired,
+    /** spacing (horizontal) */
+    spacing: number,
+    /** margin (vertical) */
+    margin: number
```

Comment:
just added default.

The name `margin` is probably illy-named. I used it to mean the vertical spacing between blocks in the same column.

what would be a better name for this quantity? @bulyonov